core/src/main/java/org/elasticsearch/rest/action/cat/RestThreadPoolAction.java

```diff
@@ -260,9 +150,31 @@ private Table buildTable(RestRequest req, ClusterStateResponse state, NodesInfoR
                     }
                 }
             }
-            for (String pool : SUPPORTED_NAMES) {
-                ThreadPoolStats.Stats poolStats = poolThreadStats.get(pool);
-                ThreadPool.Info poolInfo = poolThreadInfo.get(pool);
+            for (Map.Entry<String, ThreadPoolStats.Stats> entry : poolThreadStats.entrySet()) {
+
+                boolean matches = false;
+                for (final String threadPool : threadPools) {
+                    matches = entry.getKey().matches(threadPool);
```

Comment:
I'm trying to understand, what's the inconsistency? Is there a behavior inconsistency, or just a code inconsistency?